### PR TITLE
Feature/logit

### LINKED ISSUE #26: Implement `logit`
[Logit function](https://docs.scipy.org/doc/scipy-0.13.0/reference/generated/scipy.special.logit.html). `@stdlib/math/base/special/logit`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #26 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

* [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
* [x] Read and understand the [Code of Conduct][code-of-conduct].
* [x] Read and understood the [licensing terms][license].
* [x] Searched for existing issues and pull requests __before__ submitting this pull request.
* [x] Filed an issue (or an issue already existed) __prior to__ submitting this pull request.
* [x] Rebased onto latest `develop`.
* [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This pull request:

* Contains an implementation of the logit function
* Exports the logit function to the special functions namespace

## Related Issues

> Does this pull request have any related issues?

This pull request:

* resolves 
* fixes

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

---

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/master/CONTRIBUTING.md
[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/master/CODE_OF_CONDUCT.md
[license]: https://github.com/stdlib-js/stdlib/blob/master/LICENSE

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/logit/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# logit
+
+> Compute the [logit][logit] of a proportion.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/README.md]: > Compute the logit function.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/logit/README.md]: Perhaps include a quick intro, explaining what the logit function is; e.g., the inverse of the logistic function, as well as its functional (equation) definition.

File: lib/node_modules/@stdlib/math/base/special/logit/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Usage
+
+``` javascript
+var logit = require( '@stdlib/math/base/special/logit' );
+```
+
+#### logit( p )
+
+Computes the [logit][logit] of a proportion `p`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/README.md]: > Computes the logit function.

File: lib/node_modules/@stdlib/math/base/special/logit/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+``` javascript
+var v = logit( 0.2 );
+// returns ~-1.386
+
+v = logit( 0.9 );
+// returns ~2.197
+```
+
+If `p < 0` or `p > 1`, the function will return `NaN`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/README.md]: > the function returns `NaN`.

File: lib/node_modules/@stdlib/math/base/special/logit/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/package.json:1]: This `package.json` is out-of-date. Various fields now have default values.

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Thanks for the review! Made the requested changes.

File: lib/node_modules/@stdlib/math/base/special/logit/README.md [hunk 5]
```diff
@@ -1,6 +1,26 @@
...
+<!-- <equation class="equation" label="eq:logit_function" align="center" raw="\operatorname{logit}(p)=\log \left({\frac {p}{1-p}}\right)" alt="Logit function."> -->
+
+<div class="equation" align="center" data-raw-text="\operatorname{logit}(p)=\log \left({\frac {p}{1-p}}\right)" data-equation="eq:logit_function">
+    <img src="" alt="Logit function.">
+    <br>
+</div>
+
+<!-- </equation> -->
+
+The [logit][logit] function is the inverse of the [standard logistic] function, sometimes also called the sigmoid function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/README.md]: Broken link => `[standard logistic]`

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Thanks. Link is fixed.

File: lib/node_modules/@stdlib/math/base/special/logit/test/test.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+// FIXTURES //
+
+var small = require( './fixtures/python/small.json' );
+var medium = require( './fixtures/python/medium.json' );
+var large = require( './fixtures/python/large.json' );
+
+
+// FUNCTIONS //
+
+function almostEqual( a, b, tol ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/test.js]: We have discussed this before. This definition of "almost equal" is wanting. For instance, if `a < 1` and `b < 1`, the tolerance will be too large. Recall that `eps` is the delta between `1` and the __next__ number greater than `1`. As a reminder, this delta is not constant. For smaller values, the delta is less than `eps`.

So, as a reminder, we should __not__ be using this definition. This is particularly the case where, in this definition, `a` and `b` are treated equivalently. Here, this is not true. One value is the reference value and should be the basis for determining the neighborhood of accuracy.

If this definition was used somewhere else, that is a bug.

File: lib/node_modules/@stdlib/math/base/special/logit/README.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
+# logit
+
+> Compute the [logit][logit] function.
+
+
+<section class="intro">
+
+The [logit][logit] function is defined as the logarithm of the odds `p / (1-p)`, i.e.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/README.md]: Punctuation:

> ...the odds `p / (1-p)`; i.e.,

File: lib/node_modules/@stdlib/math/base/special/logit/test/test.js [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+	expected = small.expected;
+	x = small.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = logit( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		}  else {
+			bool = almostEqual( y, expected[i], tol );
+			t.ok( bool, 'within tolerance. x: '+x[i]+'. y: '+y+'. Expected: '+expected[i]+'. Tolerance: '+tol+'.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/test.js]: This is not the actual tolerance. The actual tolerance is scaled. Another reason we should not be using a functional definition for "almost equal".

 Should also include the `delta`.

File: lib/node_modules/@stdlib/math/base/special/logit/test/test.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+	expected = large.expected;
+	x = large.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = logit( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		}  else {
+			bool = almostEqual( y, expected[i], tol );
+			t.ok( bool, 'within tolerance. x: '+x[i]+'. y: '+y+'. Expected: '+expected[i]+'. Tolerance: '+tol+'.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/test.js]: Not the actual tolerance. Should also include the `delta`.

File: lib/node_modules/@stdlib/math/base/special/logit/test/test.js [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+	expected = medium.expected;
+	x = medium.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = logit( x[i] );
+		if ( y === expected[i] ) {
+			t.equal( y, expected[i], 'x: '+x[i]+', y: '+y+', expected: '+expected[i] );
+		}  else {
+			bool = almostEqual( y, expected[i], tol );
+			t.ok( bool, 'within tolerance. x: '+x[i]+'. y: '+y+'. Expected: '+expected[i]+'. Tolerance: '+tol+'.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/test.js]: Not the actual tolerance.  Should also include the `delta`.

File: lib/node_modules/@stdlib/math/base/special/logit/test/fixtures/python/runner.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+    # Arguments
+
+    * `x`: domain
+    * `name::str`: output filename
+
+    # Examples
+
+    ``` python
+    python> x = linspace(-1000, 1000, 2001);
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/logit/test/fixtures/python/runner.py]: Not a realistic range for this particular test. Instead,

```
python> x = linspace(0.0, 1.0, 2001);
```

**Review by kgryte — CHANGES_REQUESTED** [review]

**Comment by Planeshifter** [comment]
Replaced the `almostEqual` function call by our inline check. There are quite a few instances of the wrong check still in our `develop` branch; I will fix them over the weekend.

**Review by kgryte — APPROVED** [review]

**Pull request merged**